Bitmap base address uses memory map.

DeanoBurrito opened the pull request:
My version of using the memory map to find a place to put the bitmap!
Main difference is this _mmap_determine_bitmap_region(), which will search the memory map
until it finds space. This makes no assumptions about where free memory is.
This should work with any amount of ram too.
The only limitation is that it it requires a single free region to be large enough for the bitmap.

dreamos82 commented:
Shouldn't we mark the area used for the bitmap as reserved in the bitmap itself? :D

DeanoBurrito commented:
Where's your sense of adventure haha!
Of course, added that commit, also fixed an off-by-1 issue.

I've had to disable the pmm reserving the bitmap bits when testing. Because we cant access real memory directly in the testing environment,  the pmm is using malloc() to get memory that it would normally acquire by reading the memory map (of course we cant just write to those addresses, the tests will seg fault).
Whatever malloc returns can be way above where the bitmap would normally appear (and often above the memory allocated to a testing vm), which leads to it trying to lock pages that are out of bounds, again resulting in a seg fault. Some quick testing in qemu confirms that the bitmap is indeed reserving it's pages, and works as expected.
